### PR TITLE
Add exit dumper

### PR DESCRIPTION
**Pull request by boyuan-chen**
:clipboard: Add associated issues, tickets, docs URL here.

## Overview

Describe what your Pull Request is about in a few sentences.

This PR is for the migration.

Since L2ToL1MessagePasser.sol is updated in the v3 and it's behind the proxy, we need to dump the exit message and write them to new storage to avoid the double exit in the v3.

## Changes

Describe your changes and implementation choices. More details make PRs easier to review.

- Add dumper struct and code to write data to a file

## Testing

Describe how to test your new feature/bug fix and if possible, a step by step guide on how to demo this.

The unit tests are added

### CODE REVIEW

**Review by wsdt — APPROVED** [review]

**Review by InoMurko — APPROVED** [review]

**Comment by boyuan-chen** [comment]
I bumped up go version to v1.18 so that it supports `os.CreateTemp` and `os.ReadAll`.

File: l2geth/core/vm/evm.go [hunk 1]
```diff
@@ -579,6 +583,20 @@ func (evm *EVM) bobaTuringCall(input []byte, caller common.Address, mayBlock boo
 // the necessary steps to create accounts and reverses the state in case of an
 // execution error or failed value transfer.
 func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas uint64, value *big.Int) (ret []byte, leftOverGas uint64, err error) {
+	if addr == dump.MessagePasserAddress {
+		statedumper.WriteMessage(caller.Address(), input)
+	}
+
+	if addr == dump.OvmEthAddress {
```
Comment (souradeep-das) [inline l2geth/core/vm/evm.go:590]: what is the purpose of this?

is this for getting ETH holders? if then there could be other recipients of ETH through the native transfers?

**Review by souradeep-das — COMMENTED** [review]
lgtm! 
but should we merge this to develop? given this is to be run on the replica?

File: l2geth/core/vm/evm.go (hunk 1 above)
Comment (boyuan-chen) [reply l2geth/core/vm/evm.go:590]: This is another security layer to make sure that we get the ETH holder addresses that are used to move the ETH balance from OVM_ETH contract to the standard state in the db.

Comment (boyuan-chen) [reply l2geth/core/vm/evm.go:590]: All ETH holder addresses should be fetched via: https://github.com/bobanetwork/v3-anchorage/blob/boba-chain-ops/boba-chain-ops/cmd/eth-crawler/main.go

Comment (souradeep-das) [reply l2geth/core/vm/evm.go:590]: cool!

**Pull request merged**